doc/steps.md

```diff
@@ -155,3 +155,17 @@ Replace null values by a given value in a column.
     value: "bar"
 }
 ```
+
+### 'top' step
+
+Return top N rows by group if `groups` is specified, else over full dataset
+
+```javascript
+{
+  name: 'top',
+  groups: ['foo']
+  value: 'bar',
```

Comment:
I would suggest `key` or `sortkey` instead of `value` which sounds a bit more "static" to me.

Besides, shouldn't we accept any expression here (e.g. not just a column name)? If so, may an example would be useful.

Reply:
@adimascio Hello, not sure to understand your point. `value` receives the name of the column in which to find values to be sorted before returning top rows. What kind of generic expression should we accept ?

Ok to change the parameter name but I would rather suggest something like `values_column` or `column_to_sort` ?

Reply:
I imagined something like `value: "lowercase(my_col)"` (putting aside potential)

Reply:
Hum... Sorry but I do not see the use case. We just need to receive a regular and unique column name (I say that based on existing operations that we perform in code mode)

Reply:
Basic use case in my thoughts: I want top 10 revenue contributors by country:

```javascript
{
  name: 'top'
  groups: ['country']
  value_column: 'revenue'
}
```

May you please provide an example of yours so I can better understand ?

Reply:
I don't have any precise use case, I just think of the `key` parameter of the python's `sorted` builtin which I find very convenient. I guess a small adapation of your use case would be : "I want the 10 biggest difference in revenue since last year":

```
{
  name: 'top'
  groups: ['country']
  value: 'revenue'
  key: 'abs(revenue - last_year_revenue)'
}
```

Is it too far fetched for our standard use cases?

Reply:
Ok so in such a case `value` is not pertinent anymore because you sort only based on `key` right ? I do think it is a bit overkill at least for our first milestones. I think it makes the step more complex to understand and use and it is not something that you want to do often in such a direct fashion. Actually if you want to do this, you may use a `formula` step first to compute your revenue difference, and then apply a `top` step based on the new computed column. So we are not stuck for such a use case, and I believe that in terms of data pipeline it is easier and clearer to do it in 2 steps.

I do agree on the convenience offered by powerful features in programmatic tools such as python, but I believe that for the purpose of our graphical interface we have to keep our transformation steps as straightforward and pedagogic as possible, while promoting a modular rather than parametrical  approach to data wrangling. :) Let me know if you still disagree or if I miss something.

Reply:
@adimascio would `rank_values_on` be a better candidate to replace the parameter `value`?